pdns/slavecommunicator.cc

```diff
@@ -615,7 +615,17 @@ void CommunicatorClass::suck(const DNSName &domain, const ComboAddress& remote)
     }
   }
   catch(ResolverException &re) {
-    g_log<<Logger::Error<<"Unable to AXFR zone '"<<domain<<"' from remote '"<<remote<<"' (resolver): "<<re.reason<<endl;
+    // The AXFR probably failed due to a problem on the master server. If SOA-checks against this master
+    // still succeed, we would constantly try to AXFR the zone. To avoid this, we add the zone to the list of
+    // failed slave-checks. This will suspend slave-checks (and subsequent AXFR) for this zone for some time.
+    uint64_t newCount = 1;
+    time_t now = time(0);
+    const auto failedEntry = d_failedSlaveRefresh.find(domain);
+    if (failedEntry != d_failedSlaveRefresh.end())
+      newCount = d_failedSlaveRefresh[domain].first + 1;
+    time_t nextCheck = now + std::min(newCount * d_tickinterval, (uint64_t)::arg().asNum("soa-retry-default"));
```

Comment:
It might be a bit misleading to use `soa-retry-default` here, at the very least we should document it if we do.

Reply:
Where shall we document it? AFAIK this whole incremental backoff is not documentet somewhere.

Reply:
I think https://github.com/PowerDNS/pdns/blob/master/docs/modes-of-operation.rst#slave-operation might be the right place to write a few words about this change.

Reply:
I think it would be wise to add an option to turn the backoff (failed SOA and failed AXFR) on/off. Then we can document the exact behavior next to the option